File: AgenciaTurismo/app/src/main/java/com/example/agnciadeturismo/viewmodel/PacoteViewModel.java
package com.example.agnciadeturismo.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.agnciadeturismo.data.repository.PacoteRepositoryTask;
import com.example.agnciadeturismo.model.PacoteDto;

import java.util.ArrayList;

public class PacoteViewModel extends ViewModel{

    PacoteRepositoryTask repositoryTask = new PacoteRepositoryTask();
    private MutableLiveData<ArrayList<PacoteDto>> mOfertas = new MutableLiveData();
    private MutableLiveData<ArrayList<PacoteDto>> mPacotes = new MutableLiveData();
    public LiveData pacote = mPacotes;
    public LiveData oferta = mOfertas;

    public void getAllPacotesOferta(){
        new Thread(new Runnable() {
            @Override
            public void run() {
                mOfertas.postValue(repositoryTask.getAllOfertas());
            }
        }).start();
    }

    public void getAllPacotes(int origem, int destino, int tipo){
        new Thread(new Runnable() {
            @Override
            public void run() {
                mPacotes.postValue(repositoryTask.getAllPacotes(origem, destino, tipo));
            }
        }).start();
    }
}

File: AgenciaTurismo/app/src/main/java/com/example/agnciadeturismo/presenter/viewmodel/PacoteViewModel.kt
package com.example.agnciadeturismo.presenter.viewmodel

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.agnciadeturismo.data.repository.PacoteRepositoryTask
import com.example.agnciadeturismo.model.PacoteDto
import java.util.*

class PacoteViewModel(private val pacoteRepositoryTask: PacoteRepositoryTask) : ViewModel() {

    private val TAG = "PacoteViewModel"
    private val mOfertas = MutableLiveData<ArrayList<PacoteDto>>()
    private val mPacotes = MutableLiveData<ArrayList<PacoteDto>>()
    var pacote: LiveData<ArrayList<PacoteDto>> = mPacotes
    var oferta: LiveData<ArrayList<PacoteDto>> = mOfertas

    fun getAllPacotesOferta() {
        Thread {
            try {
                mOfertas.postValue(pacoteRepositoryTask.getAllOfertas())
            }catch (ex : Exception){ Log.d(TAG, ex.toString())}
        }.start()

    }

    fun getAllPacotes(origem: Int, destino: Int, tipo: Int) {
        Thread {
            try {
                mPacotes.postValue(pacoteRepositoryTask.getAllPacotes(origem, destino, tipo))
            }catch (ex : Exception){Log.d(TAG, ex.toString())}
        }.start()
    }

    class ViewModelFactory(private val pacoteRepositoryTask: PacoteRepositoryTask) : ViewModelProvider.Factory {
        override fun <T : ViewModel?> create(modelClass: Class<T>): T {
            return modelClass.getConstructor(PacoteRepositoryTask::class.java)
                .newInstance(pacoteRepositoryTask)
        }
    }
}